### PR TITLE
Using the State Hook

### PR DESCRIPTION
**Pull request by 0tabek16**
Salom, @umidbekkarimov @turdiyev @Shoyadbek.

- Tarjima asosan `content/docs/hooks-state.md` faylida amalga oshirildi.
- `TRANSLATION.md` faylidagi bir qancha tarjimalarga oʻzgartirish kiritildi, yangi soʻz hamda birikmalar qoʻshildi.
- "Yo'riqnoma"ni "hujjatlar" deb tajima qilingani `content/headerNav.yml` faylida toʻgʻirlandi.
- `content/docs/nav.yml` asosiy tarjima sarlavhasi o'zgartirildi.
- ish olib borilgan barcha fayllarda ' belgisi ʻ ga oʻzgartirildi.

https://deploy-preview-32--uz-reactjs.netlify.com/docs/hooks-state.html

PS: PR chigal boʻlib ketganligi uchun uzr 😅

### CODE REVIEW

File: TRANSLATION.md
Comment (jkhusanov) [inline TRANSLATION.md:25]: I think this is great!

File: TRANSLATION.md
Comment (jkhusanov) [inline TRANSLATION.md]: tuyasimonYozuv qilish kerakmidi 😃

File: content/docs/hooks-state.md
Comment (jkhusanov) [inline content/docs/hooks-state.md:33]: Comment?

File: content/docs/hooks-state.md
Comment (jkhusanov) [inline content/docs/hooks-state.md]: **ga o'xshash** qismini `this.setState({ fruit: 'orange' })` dan keyin qo'yganingiz ma'qul menimcha.

File: content/docs/hooks-state.md
Comment (jkhusanov) [inline content/docs/hooks-state.md]: **chapishtirilmaydi** so'zini o'rniga **birlashtirish** yoki **qo'shish** so'zini ishlatsak ko'proq tushunarli bo'larimidi?

**Review by jkhusanov — APPROVED** [review]
Although I'm not officially a maintainer (and that's why my approval doesn't count), I was impressed by the quality of work!  Left a few comments and food for thought, great job!

File: content/docs/hooks-state.md
Comment (0tabek16) [reply content/docs/hooks-state.md:33]: Couldn't translate the "Equivalent" word, so left the original text as comment

File: content/docs/hooks-state.md
Comment (0tabek16) [reply content/docs/hooks-state.md]: e'tiborsizlik tufayli qolib ketganga o'xshaydi, sinchikovlik uchun rahmat

**Comment by jkhusanov** [comment]
I wanted to ask it before and it's more of a question for maintainers. I saw a lot of changes related to that in this PR. Do we have a consensus on `'` and `ʻ` in letters like `o', g'`, so we could all agree on, or is it more of a personal choice? Personally, I would just use a standard English keyboard because it's easier :)

File: content/docs/hooks-state.md
Comment (0tabek16) [reply content/docs/hooks-state.md]: birlashtirishga oʻzgartiraman unda

**Comment by 0tabek16** [comment]
> I wanted to ask it previously and it's more of a question for maintainers. I saw a lot of changes related to that in this PR. Do we have a consensus on `'` and `ʻ` in letters like `o', g'`, so we could all agree on, or is it more of a personal choice? Personally, I would just use a standard English keyboard because it's easier :)

You can download keyboard layout from [here](https://farhodjon.uz/projects/uzbek-typography-keyboard-layouts/). It would be greate if we follow uzbek language syntax rules

And also, I would suggest to use [this website](https://savodxon.uz/) to check for typos

**Comment by avocadowastaken** [comment]
@0tabek16 @jkhusanov thank you guys for great work! I'm waiting for at least one more review in order to merge this PR. 

cc @turdiyev @farhodjon

**Comment by fardevio** [comment]
> I wanted to ask it before and it's more of a question for maintainers. I saw a lot of changes related to that in this PR. Do we have a consensus on `'` and `ʻ` in letters like `o', g'`, so we could all agree on, or is it more of a personal choice? Personally, I would just use a standard English keyboard because it's easier :)

Hi guys, just joined to the conversation thanks to @umidbekkarimov. Leaving my two cents on this topic and will try to review some content when necessary. I made the keyboard layout mentioned in the above comment, and also made some research to understand how to write text in Uzbek correctly (especially in the latin alphabet).

As you guys know, our latin alphabet is the worst. You can read here why: [Alifbo muammosi](https://farhodjon.uz/blog/uzbek-alphabet-problems/) (sorry for providing my own source). Furthermore, the symbol is used for `oʻ` and `gʻ` isn't a standard symbol that exists in other commonly used alphabets. Why we should use [ʻOkina](https://en.wikipedia.org/wiki/ʻOkina):

1. Recently copywriters and Wikipedia contributors have agreed to use [ʻOkina](https://en.wikipedia.org/wiki/ʻOkina), not [apostrophe](https://en.wikipedia.org/wiki/Apostrophe), not [a left single quotation mark](https://en.wikipedia.org/wiki/Quotation_mark). Also Khan Academy Uzbek uses it for their content in Uzbek.
2. Okina doesn't split a word. Apostrophe or quotation marks split a word. You can use these hashtags on Telegram to make sure: #o'zbekcha, #oʻzbekcha, #o‘zbekcha.
3. Okina is only used for the letters above, nothing else. In case if we want to change the alphabet better to have an easier solution with it rather than making a complex solution to figure out how to convert these letters correctly.

I know it's hard to type. We have to use two keyboard layouts. What if we choose the cyrillic alphabet? Then we have to use two keyboard layouts for text and code.

Let me know if you have any questions.

P.S.: currently *translation* is more important than using "correct symbols".

File: TRANSLATION.md
Comment (turdiyev) [reply TRANSLATION.md]: camelCase yaxshiku, hamma narsani tarjima qilib tashlamaylik

**Pull request merged**
